src/stream_map.rs

```diff
@@ -304,6 +335,29 @@ mod tests {
         assert!(duration >= DELAY * NUM_STREAMS);
     }
 
+    #[test]
+    fn can_iter_named_streams() {
+        const N: usize = 10;
+        let mut streams = StreamMap::new(|| Delay::futures_timer(Duration::from_millis(100)), N);
+        let mut sender = Vec::with_capacity(N);
+        for i in 0..N {
+            let (tx, rx) = mpsc::channel::<()>(1);
+            streams.try_push(format!("ID{i}"), rx).unwrap();
+            sender.push(tx);
+        }
+        assert_eq!(streams.iter_typed::<mpsc::Receiver<()>>().count(), N);
+        for (i, (id, _)) in streams.iter_typed::<mpsc::Receiver<()>>().enumerate() {
+            let expect_id = format!("ID{}", N - i - 1); // Reverse order.
```

Comment:
Huh? Why are we handing them out in reverse order?

Reply:
Not sure why, but that's how the inner `SelectAll::iter` returns the the elements. Probably an implementation detail of the `SelectAll` data structure?

Reply:
Interesting. It is a `Vec` internally. Should we sort them by ID?

Reply:
We could sort them as part of the `Iterator` implementation. `itertools` has a combinator for that.

Reply:
This would be better left to the caller: callers who want a sorted iterator can use the combinator, and those who don't care don't have to pay the performance cost of sorting/comparisons (and extra temporary memory).

Documenting that the order is unspecified should be enough to inform people of the need for sorting.

Reply:
> Documenting that the order is unspecified should be enough to inform people of the need for sorting.

Makes sense to me! Wanna send a PR?